### PR TITLE
[SPARK-22362][SQL] Add unit test for Window Aggregate Functions

### PR DESCRIPTION
**Pull request by attilapiros**
## What changes were proposed in this pull request?

Improving the test coverage of window functions focusing on missing test for window aggregate functions. No new UDAF test is added as it has been tested already.

## How was this patch tested?

Only new tests were added, automated tests were executed.

### CODE REVIEW

File: sql/core/src/test/scala/org/apache/spark/sql/DataFrameWindowFunctionsSuite.scala [hunk 1]
```diff
@@ -154,6 +154,217 @@ class DataFrameWindowFunctionsSuite extends QueryTest with SharedSQLContext {
...
+        Row("i", 0.5280049792181881, -1.5000000000000013),
+        Row("j", 0.5280049792181881, -1.5000000000000013)))
+  }
+
+  test("aggregation function on invalid column") {
+    val df = Seq((1, "1")).toDF("key", "value")
+    val e = intercept[AnalysisException](
+      df.select(
+        $"key",
+        count("invalid").over(
```
Comment (smurakozi) [inline sql/core/src/test/scala/org/apache/spark/sql/DataFrameWindowFunctionsSuite.scala]: `over()` would be enough, as partition and orderBy is ignored anyway

**Review by smurakozi — COMMENTED** [review]
You could have tests attempting to feed aggregation functions with columns containing wrong datatypes (e.g. avg on String)

**Comment by squito** [comment]
Jenkins, ok to test

File: sql/core/src/test/scala/org/apache/spark/sql/DataFrameWindowFunctionsSuite.scala (hunk 1 above)
Comment (attilapiros) [reply sql/core/src/test/scala/org/apache/spark/sql/DataFrameWindowFunctionsSuite.scala]: Thanks, I will remove the unnecessary parts.

**Comment by attilapiros** [comment]
cc  @gatorsmile @hvanhovell @jiangxb1987

**Comment by jiangxb1987** [comment]
retest this please

File: sql/core/src/test/scala/org/apache/spark/sql/DataFrameWindowFunctionsSuite.scala [hunk 2]
```diff
@@ -86,6 +93,429 @@ class DataFrameWindowFunctionsSuite extends QueryTest with SharedSQLContext {
     assert(e.message.contains("requires window to be ordered"))
   }
 
+  test("aggregation and rows between") {
+    val df = Seq((1, "1"), (2, "1"), (2, "2"), (1, "1"), (2, "2")).toDF("key", "value")
```
Comment (jiangxb1987) [inline sql/core/src/test/scala/org/apache/spark/sql/DataFrameWindowFunctionsSuite.scala]: We shall also include null data.

**Comment by jiangxb1987** [comment]
We shall also cover the sql interface, you can find some example in `sql/core/src/test/resources/sql-tests/inputs/udaf.sql`

File: sql/core/src/test/scala/org/apache/spark/sql/DataFrameWindowFunctionsSuite.scala (hunk 2 above)
Comment (attilapiros) [reply sql/core/src/test/scala/org/apache/spark/sql/DataFrameWindowFunctionsSuite.scala]: This tests was removed and re-added as result of merge conflict. Now I cleaned up.

**Comment by smurakozi** [comment]
@jiangxb1987 how does your request to cover the sql interface relates to SPARK-23160? 
I assume it is to be covered in that issue, is that correct?

**Comment by attilapiros** [comment]
I have already extended sql/core/src/test/resources/sql-tests/inputs/window.sql with the missing window aggregate functions but if you would like I can move it to a different PR too.

**Comment by attilapiros** [comment]
PySpark failure must be unrelated as only unit tests are added.

**Comment by attilapiros** [comment]
cc @gatorsmile @hvanhovell @jiangxb1987

**Comment by attilapiros** [comment]
cc @gatorsmile @hvanhovell 

@jiangxb1987 your review comments are applied, is there something else I should work on regarding this PR?

**Comment by attilapiros** [comment]
gentle ping @gatorsmile @hvanhovell

**Comment by attilapiros** [comment]
ping @hvanhovell @cloud-fan

**Review by hvanhovell — COMMENTED** [review]
A few comments. Looks good overall.

**Comment by attilapiros** [comment]
gentle reminder @hvanhovell

**Comment by hvanhovell** [comment]
LGTM - merging to master. Thanks!

**Pull request closed without merging**